x/dex/keeper/grpc_query_estimate_place_limit_order.go

```diff
@@ -53,14 +47,15 @@ func (k Keeper) EstimatePlaceLimitOrder(
 		req.OrderType,
 		req.ExpirationTime,
 		req.MaxAmountOut,
-		callerAddr,
-		receiverAddr,
+		[]byte("caller"),
+		[]byte("receiver"),
 	)
 	if err != nil {
 		return nil, err
 	}
 
-	// NB: We're only using a cache context so we don't expect any writes to happen.
+	//nolint:staticcheck // Should be unnecessary but out of an abundance of caution we restore the old bankkeeper
+	k.bankKeeper = oldBk
```

Comment:
What happens if we do an early exit at the line 54 and do not put the real keeper back?
Even if you do a `defer` to retern the keeper back, since a query is not transactional, you can brake the dex keeper with 2 concurrent queries


| q1                                     | q2                                                            |
| -------------------------------------- | ------------------------------------------------------------- |
| oldBk := k.bankKeeper                  |                                                               |
| k.bankKeeper = NewSimulationBankKeeper |                                                               |
|                                        | oldBk := k.bankKeeper (actually it's NewSimulationBankKeeper) |
|                                        | k.bankKeeper = NewSimulationBankKeeper                        |
| k.bankKeeper = oldBk                   |                                                               |
|                                        | k.bankKeeper = oldBk (NewSimulationBankKeeper)                                          |

and now we have dex keeper configured with SimulationBankKeeper, dont we?

if i got it right, my point is - it's pretty dangerous to reconfigure keeper during a query